### PR TITLE
fix(analytics): Analytics race condition + Telemetry page

### LINKED ISSUE #1489: Nice to have: Show example of telemetry sent
Not so much a bug, but more a nice to have. 

I'd feel more comfortable having analytics/telemetry enabled if an example of what is sent was shown in the readme. I could go digging through source and work it out but would save time if it was provided.

I'd like to help you guys out with this (as I'm sure many others would) but need to be confident I'm only sending data I'm comfortable with.

### PR DESCRIPTION
**Pull request by 10ko**
**What this PR does / why we need it**:
This PR introduces two changes:
- Detail page about telemetry with samples of data we collect.
- Workaround for a race condition in the segment client which would cause the first event to potentially never been sent to Segment. See [here](https://github.com/segmentio/analytics-node/issues/219).

**Which issue(s) this PR fixes**:

Fixes #1489

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Telemetry
+
+Whenever you run a Garden command we collect anonymized telemetry and send it to [Segment.io](https://segment.io).
+It helps us understand how our users use garden and aids our decision process when prioritizing new features, bug fixing, etc.
```
Comment (edvald) [inline docs/telemetry.md]: garden -> Garden

File: docs/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+We put great effort in making sure we only collect completely anonymized data: we use random generated uuids for identifying users and we hash sensitive information using SHA-256.
```
Comment (edvald) [inline docs/telemetry.md]: : we -> : We
uuids -> UUIDs
users -> users and projects

File: docs/telemetry.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+## Types of events
+
+Below you can find examples for each one of the events we currently collect.
```
Comment (edvald) [inline docs/telemetry.md]: Maybe safer to just say "examples of events we collect", unless we commit to always update this when we add new events or properties. If we go for the latter I'd suggest automating this, similar to how we generate reference docs, but imo that would be good for a later iteration.

File: docs/telemetry.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+**Note:** events are batched before being sent to our data processor. This means you shouldn't worry about Garden making dozens of HTTP requests and affecting commands performance.
```
Comment (edvald) [inline docs/telemetry.md]: I'm not sure this is necessary to mention. Seems like a table-stakes thing to do in our implementation :)

File: docs/telemetry.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+### Identify
+
+Collected the first time a Garden command is run on a machine.
```
Comment (edvald) [inline docs/telemetry.md]: Collected -> Sent (easier to read that way)

File: docs/telemetry.md [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    system: {
+      gardenVersion: '0.10.16',
+      platform: 'darwin',
+      platformVersion: '18.7.0'
+    }
+  }
+}
+```
+
+## Update your telemetry preferences
```
Comment (edvald) [inline docs/telemetry.md]: Update -> Updating

File: docs/telemetry.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+If you would like to update your analytics settings, please run (from within a project):
```
Comment (edvald) [inline docs/telemetry.md]: Wasn't the "from within a project" requirement removed recently?

Comment (eysi09) [reply docs/telemetry.md]: 'twas

File: garden-service/src/analytics/analytics.ts [hunk 8]
```diff
@@ -319,15 +320,20 @@ export class AnalyticsHandler {
...
+            if (err && this.log) {
+              this.log.debug(`Error sending tracking event: ${err}`)
+            }
+            resolve(true)
+          })
         })
       }
 
-      return trackToRemote(segmentEvent)
+      return await trackToRemote(segmentEvent)
```
Comment (eysi09) [inline garden-service/src/analytics/analytics.ts]: Should we `debug` (or maybe `silly`) log the data we're sending?

Comment (edvald) [reply garden-service/src/analytics/analytics.ts]: A silly log would make sense, yes.

Also, `return await` is in most cases redundant. Just noticing that now. No harm, but it means creating an additional Promise that is resolved at the same time as the other one.

File: docs/telemetry.md (hunk 2 above)
Comment (10ko) [reply docs/telemetry.md]: We hash the projectId so technically we don't use UUIDs to identify projects.

File: garden-service/src/analytics/analytics.ts (hunk 8 above)
Comment (10ko) [reply garden-service/src/analytics/analytics.ts]: Will fix and add `silly` log

**Comment by 10ko** [comment]
Fixed.

**Comment by edvald** [comment]
There are still a couple of unresolved comments, otherwise looks good.

**Comment by eysi09** [comment]
> There are still a couple of unresolved comments, otherwise looks good.

I'll go ahead and approve this then since those comments have already been resolved. So that we can into this release.

**Pull request merged**